core/src/display_object/container.rs

```diff
@@ -597,13 +602,49 @@ impl<'gc> ChildContainer<'gc> {
     ///
     /// This returns `true` if the child was successfully removed, and `false`
     /// if no list alterations were made.
-    fn remove_child_from_render_list(&mut self, child: DisplayObject<'gc>) -> bool {
-        let render_list_position = self
+    ///
+    /// This must be called *before* setting the child's parent field to `None`.
+    /// Note: This cannot be a normal method that takes &self, since we need to call
+    /// `parent.object2()` on our own `DisplayObject` (which is borrowed mutably
+    /// by `raw_container_mut`)
+    fn remove_child_from_render_list(
+        container: DisplayObjectContainer<'gc>,
+        child: DisplayObject<'gc>,
+        context: &mut UpdateContext<'_, 'gc>,
+    ) -> bool {
+        let mut this = container.raw_container_mut(context.gc_context);
+
+        let render_list_position = this
             .render_list
             .iter()
             .position(|x| DisplayObject::ptr_eq(*x, child));
         if let Some(position) = render_list_position {
-            self.render_list_mut().remove(position);
+            this.render_list_mut().remove(position);
+            drop(this);
+
+            // Only set the parent's field to 'null' if the child was not placed/modified
+            // on the render list by AVM2 code.
+            if !child.placed_by_script() {
+                let parent = child.parent().expect(
+                    "Parent must be removed *after* calling `remove_child_from_render_list`",
+                );
+                if child.has_explicit_name() {
+                    if let Avm2Value::Object(mut parent_obj) = parent.object2() {
+                        let mut activation = Avm2Activation::from_nothing(context.reborrow());
+                        parent_obj
+                            .set_public_property(child.name(), Avm2Value::Null, &mut activation)
+                            .unwrap_or_else(|err| {
+                                panic!(
```

Comment:
the sibling function that sets the child, doesn't panic here, it just logs and continues.  We probably don't want a panic here, since this can invoke a user setter that can throw.